tools/pull/pull.cpp

```diff
@@ -0,0 +1,68 @@
+#include "arg.h"
+#include "common.h"
+#include "log.h"
+
+#include <cstdio>
+#include <string>
+
+static void print_usage(int, char ** argv) {
+    LOG("Usage: %s [options]\n", argv[0]);
+    LOG("\n");
+    LOG("Download models from HuggingFace or Docker Hub\n");
+    LOG("\n");
+    LOG("Options:\n");
+    LOG("  -h, --help                show this help message and exit\n");
+    LOG("  -hf, -hfr, --hf-repo REPO download model from HuggingFace repo\n");
+    LOG("                            format: <user>/<model>[:<quant>]\n");
+    LOG("                            example: microsoft/DialoGPT-medium\n");
+    LOG("  -dr, --docker-repo REPO   download model from Docker Hub\n");
+    LOG("                            format: [<repo>/]<model>[:<quant>]\n");
+    LOG("                            example: gemma3\n");
+    LOG("  --hf-token TOKEN          HuggingFace token for private repos\n");
+    LOG("\n");
+    LOG("Examples:\n");
+    LOG("  %s -hf microsoft/DialoGPT-medium\n", argv[0]);
+    LOG("  %s -dr gemma3\n", argv[0]);
+    LOG("  %s -hf microsoft/DialoGPT-medium\n", argv[0]);
+    LOG("\n");
+}
+
+int main(int argc, char ** argv) {
+    common_params params;
+
+    // Set up argument parsing context
+    auto ctx = common_params_parser_init(params, LLAMA_EXAMPLE_COMMON, print_usage);
+
+    // Parse command line arguments
+    if (!common_params_parse(argc, argv, params, LLAMA_EXAMPLE_COMMON)) {
```

Comment:
Should this perhaps be:
```suggestion
    if (!common_params_parse(argc, argv, params, LLAMA_EXAMPLE_COMMON, print_usage)) {
```
And then the previous `common_params_parser_init` removed as `common_params_parse` will also call `common_params_parser_init` but with the default `print_usage` (nullptr) as it is not passed here.

Currently, the `--help` option does not print the usage information for the pull tool.